Assets/MixedRealityToolkit-SDK/Features/Diagnostics/MixedRealityDiagnosticsManager.cs

```diff
@@ -0,0 +1,228 @@
+        {
+            eventData = new DiagnosticsEventData(EventSystem.current);
+
+            Visible = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.Visible;
+            ShowCpu = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.ShowCpu;
+            ShowFps = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.ShowFps;
+            ShowMemory = MixedRealityManager.Instance.ActiveProfile.DiagnosticsProfile.ShowMemory;
+
+            if (Visible)
+            {
+                DiagnosticVisualization.SetActive(Visible);
+            }
+
+            RaiseDiagnosticsChanged();
+        }
+
+        /// <inheritdoc />
+        public override void Reset()
+        {
+            base.Reset();
+            InitializeInternal();
+        }
+
+        public override void Destroy()
+        {
+            base.Destroy();
+
+            if (Application.isPlaying)
+            {
+                if (diagnosticVisualization != null)
+                {
+                    if (Application.isEditor)
+                    {
+                        Object.DestroyImmediate(diagnosticVisualization);
+                    }
+                    else
+                    {
+                        Object.Destroy(diagnosticVisualization);
+                    }
+
+                    diagnosticVisualization = null;
+                }
+
+                visible = false;
+                showCpu = false;
+                showFps = false;
+                showMemory = false;
+
+                RaiseDiagnosticsChanged();
+            }
+        }
+
+        private void RaiseDiagnosticsChanged()
+        {
+            eventData.Initialize(this,
+                visible: Visible,
+                showCpu: ShowCpu,
+                showFps: ShowFps,
+                showMemory: ShowMemory
+                );
+
+            HandleEvent(eventData, OnDiagnosticsChanged);
+        }
+
+        /// <summary>
+        /// Event sent whenever the boundary visualization changes.
+        /// </summary>
+        private static readonly ExecuteEvents.EventFunction<IMixedRealityDiagnosticsHandler> OnDiagnosticsChanged =
+            delegate (IMixedRealityDiagnosticsHandler handler, BaseEventData eventData)
+            {
+                DiagnosticsEventData diagnosticsEventsData = ExecuteEvents.ValidateEventData<DiagnosticsEventData>(eventData);
+                handler.OnDiagnosticSettingsChanged(diagnosticsEventsData);
+            };
+
+        #endregion IMixedRealityManager
+
+        #region IMixedRealityDiagnosticsManager
+        private bool visible;
+
+        /// <inheritdoc />
+        public bool Visible
+        {
+            get
+            {
+                return visible;
+            }
+
+            set
+            {
+                if (value != visible)
+                {
+                    visible = value;
+                    DiagnosticVisualization?.SetActive(value);
```

Comment:
here.

Reply:
Good point. It was originally designed for if going from true > false, but that would imply that the default value (false) was switched to true before then. I'll remove the check here.